packages/eslint-plugin/src/rules/padding-line-between-statements.ts

```diff
@@ -0,0 +1,783 @@
+import {
+  AST_NODE_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+/**
+ * This rule is a replica of padding-line-between-statements.
+ *
+ * Ideally we would want to extend the rule support typescript specific support.
+ * But since not all the state is exposed by the eslint and eslint has frozen stylistic rules,
+ * (see - https://eslint.org/blog/2020/05/changes-to-rules-policies for details.)
+ * we are forced to re-implement the rule here.
+ *
+ * We have tried to keep the implementation as close as possible to the eslint implementation, to make
+ * patching easier for future contributors.
+ *
+ * Reference rule - https://github.com/eslint/eslint/blob/master/lib/rules/padding-line-between-statements.js
+ */
+
+//------------------------------------------------------------------------------
+// Local Types
+//------------------------------------------------------------------------------
+
```

Comment:
nit - delete this as we don't do this sectioning in our codebase

Reply:
Delete these in the main rule file, have left it in the test file, Since it helps to demarcate the various tests in 4000 LOC